### ISSUE TITLE
Auto frameskip is broken in recent git builds

### ISSUE THREAD

**Issue by Stizzie** [hrydgard/ppsspp#15525]
In recent git builds frameskipping produces super choppy render, like 10-15fps it feels like.

**Comment by ANR2ME** [comment]
The devs will need more info, like what device are you using? which games were affected?
And are you sure it only affected by auto-frameskipping and not a combination of other settings (ie. custom CPU clock, specific renderer backend, etc) ?
Have you tried resetting the settings to default and only enable/disable auto-frameskipping?

**Comment by ANR2ME** [comment]
I was hoping i'm getting what no.2 is saying:
> Let's say the game is a 30 FPS game. Auto frameskip is on, and # of frames is set to 8. However, the device is powerful enough to run it at 60 FPS. Because it is unnecessary to skip any frames, the target FPS remains 30.

But apparently, regardless whether i enable/disable auto-frameskipping, the same thing will be shown, the same skipping effect (jumpy movements), and the thing on FPS info too (set to both) if i set "Frame skipping" to a number. 
It's been like this for quite some times i think, not just recently, but since my hardware is capable to run full speed, it doesn't really bother me much (usually i just set Frame skipping to Off).

For example on GT Main menu, with "Frame skipping" Off, and Auto-frameskipping Off, Movements are smooth
![image](https://user-images.githubusercontent.com/7974720/167585755-69f02479-6564-4114-aad3-212d3e70fbf6.png)

With "Frame skipping" to 8, and Auto-frameskipping On/Off (doesn't matter), Movements are choppy/jumpy
![image](https://user-images.githubusercontent.com/7974720/167586210-9292b3c2-435c-4e40-9407-3612efdc204b.png)

**Comment by Gamemulatorer** [comment]
Maybe related to this
https://github.com/hrydgard/ppsspp/issues/15276

**Comment by ANR2ME** [comment]
> Maybe related to this #15276

That issue is related to flickering, while this issue is related to suspectedly broken frameskipping:

On dev builds "Frame skipping" 8 will cause the screen to be updated once per second, causing it to looked like a slide show, while on v1.12.3 or older version frameskips are smoother (smaller skips instead of large skip).

There is also a oddity where auto-frameskipping not to have any difference being On/Off, which is suspected to be broken too.

**Comment by unknownbrackets** [comment]
I'm guessing this must be related to some other setting, then.  I don't experience this, though I don't have Gran Turismo I don't think.

Games I quickly checked with auto frameskip and frameskip set to 8:
 * Legend of Heroes 3: Song of the Ocean
 * Brave Story
 * God of War: Chains of Olympus Demo
 * Crisis Core
 * Ridge Racer (I did get a hitch that caused a frameskip, probably due to a shader or something, but it was momentary.)
 * LittleBigPlanet

Tried with OpenGL, Vulkan, and Direct3D 11.

Maybe there's something specific with detection of Gran Turismo (there definitely is with Danganronpa 2), but if you're seeing this widely it must be something else.

-[Unknown]

**Comment by unknownbrackets** [comment]
Don't have time to test now, but it seems likely that real time clock sync (which generally exists to slow things down so time lines up with emulation as much as possible) and frameskip could fight with each other.  It probably doesn't make sense for them to be enabled concurrently.  Maybe that's what you're experiencing, and I do think that we fixed bugs in real time clock sync around the versions you noted.

-[Unknown]

**Comment by ANR2ME** [comment]
I tried to reset the setting to defau;t, and you're right!
Auto-frameskipping worked properly now with "real time clock sync" disabled, didn't know they can't be enabled together before.

But the issue where the screen only updated once per second like a slide show is still there in dev builds, is this an intended behavior? because v1.12.3 release and older version wasn't like this

**Comment by unknownbrackets** [comment]
In Fat Princess, ForceMax60FPS seems to be interacting poorly with frameskip.  In theory, frameskip shouldn't change game timing so this probably is a bug, like #15276 or really #14315.

-[Unknown]

**Comment by Gamemulatorer** [comment]
Not directly related to this issue but skip-buffer effects (non-buffered rendering) is broken since https://github.com/hrydgard/ppsspp/commit/1dead5f095987c812f6d104db5618e7658dcdf1b


https://user-images.githubusercontent.com/37603562/169637634-9e7ef4b7-8718-47da-a079-bf41d9406449.mp4

**Comment by RomRider** [comment]
Same issue on pi4, used to run most of the games quite smoothly with auto frameskip (most of the time 60fps and skipping when required), now it's super choppy, auto frameskip enabled makes a game render at 7 fps when it can actually render at 60fps when number of frame skip is set to off (eg: wipeout pure)

**Comment by ANR2ME** [comment]
> Same issue on pi4, used to run most of the games quite smoothly with auto frameskip (most of the time 60fps and skipping when required), now it's super choppy, auto frameskip enabled makes a game render at 7 fps when it can actually render at 60fps when number of frame skip is set to off (eg: wipeout pure)

Make sure "real clock sync" is disabled on System setting when using Auto-frameskipping

**Comment by unknownbrackets** [comment]
It'd really help if you could try different builds of PPSSPP and find the first build with the problem, and the last build without it.

If you're compiling, you can use this:
https://github.com/hrydgard/ppsspp/wiki/How-to-bisect-to-find-what-broke-a-game

-[Unknown]

**Comment by RomRider** [comment]
> Make sure "real clock sync" is disabled on System setting when using Auto-frameskipping

It is, it's even grayed out.

> It'd really help if you could try different builds of PPSSPP and find the first build with the problem, and the last build without it.

I'm running Batocera so kind of hard to do that.

It's somewhere between v1.12.3 and commit 221976ff4057e7d9a7a00e4bb94e9656e9f8695a

**Comment by unknownbrackets** [comment]
> It's somewhere between v1.12.3 and commit [221976f](https://github.com/hrydgard/ppsspp/commit/221976ff4057e7d9a7a00e4bb94e9656e9f8695a)

There's 1200 changes between those two, so unfortunately it's just a big haystack of possibilities.

-[Unknown]

**Comment by hrydgard** [comment]
I am struggling to reproduce this... Can you supply your ppsspp.ini, or very specific settings?

**Comment by Gamemulatorer** [comment]
Yes auto frameskip is not smooth as before that stable in fps 50/60,55/60,60/60 for example in Tekken 6
![Screenshot_2022-07-11-20-49-56-238_org ppsspp ppsspp](https://user-images.githubusercontent.com/37603562/178268367-e71d5734-8ef7-43f7-a118-91497ebf04d3.jpg)

**Comment by hrydgard** [comment]
OK I've messed around with autoframeskip and Wipeout on my Rasp 4 (desktop mode), and yeah, it's indeed very strangely choppy. Frame times are ridiculously uneven, first two frames in quick succession, then a gap until the next one. Also when running with much lower frameskip or no frameskip, it's like every other frame is much faster than the other, which might have strange downstream effects. 

Either way, now that I have some kind of repro, I'm can bisect to see if this broke since 1.12, and I will soon give it a shot.

**Comment by hrydgard** [comment]
Hm, went all the way back to 1.12.3 and the rasp4 is similary choppy, so that problem seems unrelated (maybe it's throttling up and down or something due to my possibly inadequate phone charger as a power supply)

**Comment by hrydgard** [comment]
Alright, I've tried Tekken on a Samsung Galaxy S3, and at 8 autoframeskip it actually runs full speed although extremely choppy. It's just as choppy in both 1.12.3 and the current version though, I really can't tease out any substantial difference that I can try to find the cause of...

**Comment by hrydgard** [comment]
And now GT as well. I'm seriously struggling to find a reproduction of a case where auto frameskip behaves worse in the latest builds than in 1.12.3. 

Please, anybody who cares about this, make a reproducible comparison where 1.12.3 is obviously better, and I'll try to get something in. Otherwise the release is gonna be as-is, and any issues will be fixed afterwards.

**Comment by Gamemulatorer** [comment]
Hrydgard I cannot reproduce this issue anymore after updating my phone.

@Stizzie can you still reproduce this issue?

**Comment by ANR2ME** [comment]
Apparently auto-frameskip 8 doesn't worked well with IR interpreter :( it rarely gets 60 FPS on this screen
![image](https://user-images.githubusercontent.com/7974720/181766196-98c4fbf7-a9d3-44ca-b40d-85c2ba5a3f9c.png)
While it get a stable 60 FPS without frameskip

PS: i have TurboBoost disabled, thus my CPU clock is capped at 1.8 ghz

**Comment by unknownbrackets** [comment]
I don't think we've seen anything about auto frameskip being broken for some months now.  Is it possible whatever issue has been fixed?  I've never reproduced any issue with it.

Interpreter and IR interpreter will naturally make frameskip or auto frameskip not work as well, because frameskip only skips drawing.  The CPU still has to run, so if the CPU emulation is the slowest thing - FPS won't be any better by skipping drawing.

-[Unknown]
